### ISSUE TITLE
bookinfo does not work with 0618 build

### ISSUE THREAD

**Issue by gyliu513** [istio/istio#6460]
1) Get daily build from https://gcsweb.istio.io/gcs/istio-release-pipeline-data/daily-build/master-20180618-09-15/
2) Install istio and bookinfo
3) Use NodePort access ingressgateway productpage, failed.

```
root@gyliu-c21:~/istio-master-20180618-09-15# curl 172.16.250.141:31380/productpage
upstream connect error or disconnect/reset before headers
```

Productpage proxy log:
```
2018-06-21T07:22:22.551113Z	info	Envoy command: [-c /etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 45 --parent-shutdown-time-s 60 --service-cluster productpage --service-node sidecar~200.1.44.157~productpage-v1-c89f489f4-z4jfg.default~default.svc.cluster.local --max-obj-name-len 189 -l warn --v2-config-only]
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:182] initializing epoch 0 (hot restart version=10.200.16384.256.options=capacity=16384, num_slots=8209 hash=228984379728933363 size=4882536)
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:184] statically linked extensions:
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:186]   access_loggers: envoy.file_access_log,envoy.http_grpc_access_log
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:189]   filters.http: envoy.buffer,envoy.cors,envoy.ext_authz,envoy.fault,envoy.filters.http.rbac,envoy.grpc_http1_bridge,envoy.grpc_json_transcoder,envoy.grpc_web,envoy.gzip,envoy.health_check,envoy.http_dynamo_filter,envoy.ip_tagging,envoy.lua,envoy.rate_limit,envoy.router,envoy.squash,istio_authn,jwt-auth,mixer
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:192]   filters.listener: envoy.listener.original_dst,envoy.listener.proxy_protocol,envoy.listener.tls_inspector
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:195]   filters.network: envoy.client_ssl_auth,envoy.echo,envoy.ext_authz,envoy.http_connection_manager,envoy.mongo_proxy,envoy.ratelimit,envoy.redis_proxy,envoy.tcp_proxy,mixer
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:197]   stat_sinks: envoy.dog_statsd,envoy.metrics_service,envoy.statsd
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:199]   tracers: envoy.dynamic.ot,envoy.lightstep,envoy.zipkin
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:202]   transport_sockets.downstream: alts,envoy.transport_sockets.capture,raw_buffer,tls
[2018-06-21 07:22:22.573][50][info][main] external/envoy/source/server/server.cc:205]   transport_sockets.upstream: alts,envoy.transport_sockets.capture,raw_buffer,tls
[2018-06-21 07:22:22.579][50][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:235] gRPC config stream closed: 14, no healthy upstream
[2018-06-21 07:22:22.579][50][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:39] Unable to establish new stream
[2018-06-21 07:22:22.579][50][info][config] external/envoy/source/server/configuration_impl.cc:53] loading 0 listener(s)
[2018-06-21 07:22:22.579][50][info][config] external/envoy/source/server/configuration_impl.cc:87] loading tracing configuration
[2018-06-21 07:22:22.579][50][info][config] external/envoy/source/server/configuration_impl.cc:96]   loading tracing driver: envoy.zipkin
[2018-06-21 07:22:22.579][50][info][config] external/envoy/source/server/configuration_impl.cc:109] loading stats sink configuration
[2018-06-21 07:22:22.579][50][critical][main] external/envoy/source/server/server.cc:77] error initializing configuration '/etc/istio/proxy/envoy-rev0.json': malformed IP address: istio-statsd-prom-bridge
[2018-06-21 07:22:22.580][50][info][main] external/envoy/source/server/server.cc:436] exiting
2018-06-21T07:22:22.583971Z	warn	Epoch 0 terminated with an error: exit status 1
2018-06-21T07:22:22.584013Z	warn	Aborted all epochs
2018-06-21T07:22:22.584073Z	info	Epoch 0: set retry delay to 25.6s, budget to 2
2018-06-21T07:22:48.184277Z	info	Reconciling configuration (budget 2)
2018-06-21T07:22:48.184430Z	info	Epoch 0 starting
2018-06-21T07:22:48.185555Z	info	Envoy command: [-c /etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 45 --parent-shutdown-time-s 60 --service-cluster productpage --service-node sidecar~200.1.44.157~productpage-v1-c89f489f4-z4jfg.default~default.svc.cluster.local --max-obj-name-len 189 -l warn --v2-config-only]
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:182] initializing epoch 0 (hot restart version=10.200.16384.256.options=capacity=16384, num_slots=8209 hash=228984379728933363 size=4882536)
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:184] statically linked extensions:
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:186]   access_loggers: envoy.file_access_log,envoy.http_grpc_access_log
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:189]   filters.http: envoy.buffer,envoy.cors,envoy.ext_authz,envoy.fault,envoy.filters.http.rbac,envoy.grpc_http1_bridge,envoy.grpc_json_transcoder,envoy.grpc_web,envoy.gzip,envoy.health_check,envoy.http_dynamo_filter,envoy.ip_tagging,envoy.lua,envoy.rate_limit,envoy.router,envoy.squash,istio_authn,jwt-auth,mixer
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:192]   filters.listener: envoy.listener.original_dst,envoy.listener.proxy_protocol,envoy.listener.tls_inspector
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:195]   filters.network: envoy.client_ssl_auth,envoy.echo,envoy.ext_authz,envoy.http_connection_manager,envoy.mongo_proxy,envoy.ratelimit,envoy.redis_proxy,envoy.tcp_proxy,mixer
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:197]   stat_sinks: envoy.dog_statsd,envoy.metrics_service,envoy.statsd
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:199]   tracers: envoy.dynamic.ot,envoy.lightstep,envoy.zipkin
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:202]   transport_sockets.downstream: alts,envoy.transport_sockets.capture,raw_buffer,tls
[2018-06-21 07:22:48.209][56][info][main] external/envoy/source/server/server.cc:205]   transport_sockets.upstream: alts,envoy.transport_sockets.capture,raw_buffer,tls
[2018-06-21 07:22:48.215][56][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:235] gRPC config stream closed: 14, no healthy upstream
[2018-06-21 07:22:48.215][56][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:39] Unable to establish new stream
[2018-06-21 07:22:48.215][56][info][config] external/envoy/source/server/configuration_impl.cc:53] loading 0 listener(s)
[2018-06-21 07:22:48.215][56][info][config] external/envoy/source/server/configuration_impl.cc:87] loading tracing configuration
[2018-06-21 07:22:48.215][56][info][config] external/envoy/source/server/configuration_impl.cc:96]   loading tracing driver: envoy.zipkin
[2018-06-21 07:22:48.216][56][info][config] external/envoy/source/server/configuration_impl.cc:109] loading stats sink configuration
[2018-06-21 07:22:48.216][56][critical][main] external/envoy/source/server/server.cc:77] error initializing configuration '/etc/istio/proxy/envoy-rev0.json': malformed IP address: istio-statsd-prom-bridge
2018-06-21T07:22:48.221240Z	warn	Epoch 0 terminated with an error: exit status 1
2018-06-21T07:22:48.221305Z	warn	Aborted all epochs
2018-06-21T07:22:48.221384Z	info	Epoch 0: set retry delay to 51.2s, budget to 1
[2018-06-21 07:22:48.217][56][info][main] external/envoy/source/server/server.cc:436] exiting
2018-06-21T07:23:39.421607Z	info	Reconciling configuration (budget 1)
2018-06-21T07:23:39.421806Z	info	Epoch 0 starting
2018-06-21T07:23:39.422683Z	info	Envoy command: [-c /etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 45 --parent-shutdown-time-s 60 --service-cluster productpage --service-node sidecar~200.1.44.157~productpage-v1-c89f489f4-z4jfg.default~default.svc.cluster.local --max-obj-name-len 189 -l warn --v2-config-only]
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:182] initializing epoch 0 (hot restart version=10.200.16384.256.options=capacity=16384, num_slots=8209 hash=228984379728933363 size=4882536)
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:184] statically linked extensions:
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:186]   access_loggers: envoy.file_access_log,envoy.http_grpc_access_log
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:189]   filters.http: envoy.buffer,envoy.cors,envoy.ext_authz,envoy.fault,envoy.filters.http.rbac,envoy.grpc_http1_bridge,envoy.grpc_json_transcoder,envoy.grpc_web,envoy.gzip,envoy.health_check,envoy.http_dynamo_filter,envoy.ip_tagging,envoy.lua,envoy.rate_limit,envoy.router,envoy.squash,istio_authn,jwt-auth,mixer
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:192]   filters.listener: envoy.listener.original_dst,envoy.listener.proxy_protocol,envoy.listener.tls_inspector
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:195]   filters.network: envoy.client_ssl_auth,envoy.echo,envoy.ext_authz,envoy.http_connection_manager,envoy.mongo_proxy,envoy.ratelimit,envoy.redis_proxy,envoy.tcp_proxy,mixer
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:197]   stat_sinks: envoy.dog_statsd,envoy.metrics_service,envoy.statsd
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:199]   tracers: envoy.dynamic.ot,envoy.lightstep,envoy.zipkin
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:202]   transport_sockets.downstream: alts,envoy.transport_sockets.capture,raw_buffer,tls
[2018-06-21 07:23:39.444][61][info][main] external/envoy/source/server/server.cc:205]   transport_sockets.upstream: alts,envoy.transport_sockets.capture,raw_buffer,tls
[2018-06-21 07:23:39.451][61][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:235] gRPC config stream closed: 14, no healthy upstream
[2018-06-21 07:23:39.451][61][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:39] Unable to establish new stream
[2018-06-21 07:23:39.451][61][info][config] external/envoy/source/server/configuration_impl.cc:53] loading 0 listener(s)
[2018-06-21 07:23:39.451][61][info][config] external/envoy/source/server/configuration_impl.cc:87] loading tracing configuration
[2018-06-21 07:23:39.451][61][info][config] external/envoy/source/server/configuration_impl.cc:96]   loading tracing driver: envoy.zipkin
[2018-06-21 07:23:39.451][61][info][config] external/envoy/source/server/configuration_impl.cc:109] loading stats sink configuration
[2018-06-21 07:23:39.451][61][critical][main] external/envoy/source/server/server.cc:77] error initializing configuration '/etc/istio/proxy/envoy-rev0.json': malformed IP address: istio-statsd-prom-bridge
[2018-06-21 07:23:39.452][61][info][main] external/envoy/source/server/server.cc:436] exiting
2018-06-21T07:23:39.455232Z	warn	Epoch 0 terminated with an error: exit status 1
2018-06-21T07:23:39.455273Z	warn	Aborted all epochs
2018-06-21T07:23:39.455324Z	info	Epoch 0: set retry delay to 1m42.4s, budget to 0
2018-06-21T07:25:21.855498Z	info	Reconciling configuration (budget 0)
2018-06-21T07:25:21.855620Z	info	Epoch 0 starting
2018-06-21T07:25:21.856602Z	info	Envoy command: [-c /etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 45 --parent-shutdown-time-s 60 --service-cluster productpage --service-node sidecar~200.1.44.157~productpage-v1-c89f489f4-z4jfg.default~default.svc.cluster.local --max-obj-name-len 189 -l warn --v2-config-only]
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:182] initializing epoch 0 (hot restart version=10.200.16384.256.options=capacity=16384, num_slots=8209 hash=228984379728933363 size=4882536)
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:184] statically linked extensions:
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:186]   access_loggers: envoy.file_access_log,envoy.http_grpc_access_log
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:189]   filters.http: envoy.buffer,envoy.cors,envoy.ext_authz,envoy.fault,envoy.filters.http.rbac,envoy.grpc_http1_bridge,envoy.grpc_json_transcoder,envoy.grpc_web,envoy.gzip,envoy.health_check,envoy.http_dynamo_filter,envoy.ip_tagging,envoy.lua,envoy.rate_limit,envoy.router,envoy.squash,istio_authn,jwt-auth,mixer
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:192]   filters.listener: envoy.listener.original_dst,envoy.listener.proxy_protocol,envoy.listener.tls_inspector
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:195]   filters.network: envoy.client_ssl_auth,envoy.echo,envoy.ext_authz,envoy.http_connection_manager,envoy.mongo_proxy,envoy.ratelimit,envoy.redis_proxy,envoy.tcp_proxy,mixer
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:197]   stat_sinks: envoy.dog_statsd,envoy.metrics_service,envoy.statsd
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:199]   tracers: envoy.dynamic.ot,envoy.lightstep,envoy.zipkin
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:202]   transport_sockets.downstream: alts,envoy.transport_sockets.capture,raw_buffer,tls
[2018-06-21 07:25:21.880][67][info][main] external/envoy/source/server/server.cc:205]   transport_sockets.upstream: alts,envoy.transport_sockets.capture,raw_buffer,tls
[2018-06-21 07:25:21.885][67][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:235] gRPC config stream closed: 14, no healthy upstream
[2018-06-21 07:25:21.885][67][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:39] Unable to establish new stream
[2018-06-21 07:25:21.885][67][info][config] external/envoy/source/server/configuration_impl.cc:53] loading 0 listener(s)
[2018-06-21 07:25:21.885][67][info][config] external/envoy/source/server/configuration_impl.cc:87] loading tracing configuration
[2018-06-21 07:25:21.885][67][info][config] external/envoy/source/server/configuration_impl.cc:96]   loading tracing driver: envoy.zipkin
[2018-06-21 07:25:21.886][67][info][config] external/envoy/source/server/configuration_impl.cc:109] loading stats sink configuration
[2018-06-21 07:25:21.886][67][critical][main] external/envoy/source/server/server.cc:77] error initializing configuration '/etc/istio/proxy/envoy-rev0.json': malformed IP address: istio-statsd-prom-bridge
[2018-06-21 07:25:21.886][67][info][main] external/envoy/source/server/server.cc:436] exiting
2018-06-21T07:25:21.889321Z	warn	Epoch 0 terminated with an error: exit status 1
2018-06-21T07:25:21.889358Z	warn	Aborted all epochs
2018-06-21T07:25:21.889398Z	error	Permanent error: budget exhausted trying to fulfill the desired configuration
2018-06-21T07:25:21.889404Z	error	cannot start the proxy with the desired configuration
```

/cc @rshriram @frankbu

**Comment by ymesika** [comment]
I don't see https://github.com/istio/istio/pull/6349 in this daily build.
Probably the reason.

**Comment by gyliu513** [comment]
Thanks @ymesika , from the log, your suspect is right ;-)
```
[2018-06-21 07:25:21.886][67][critical][main] external/envoy/source/server/server.cc:77] error initializing configuration '/etc/istio/proxy/envoy-rev0.json': malformed     IP address: istio-statsd-prom-bridge
```

Fixed in #6349
